### ISSUE TITLE
YOLO26n: Validation metrics improved over YOLOv8n but video inference performance is significantly degraded

### ISSUE THREAD

**Issue by Abdullah-Ory** [ultralytics/ultralytics#23510]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I trained YOLO26n and YOLOv8n on the same custom dataset using identical training parameters (same epochs, batch size, image size, augmentations, etc.). The YOLO26n model shows improved validation metrics (higher mAP, precision, recall) compared to YOLOv8n a little bit not that improvement.
However, when I run inference using the best.pt weights on video, YOLO26n performs significantly worse than YOLOv8n while YOLOv8b performs very excellent. This behavior is consistent across multiple test videos.
Any ideas why? 

### Environment

python 3.12.12

### Minimal Reproducible Example

code can not be provided due to the confidential organization 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Abdullah-Ory, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

This is an automated response 🤖—an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. We understand you can’t share confidential code; an MRE can still be created by reducing to a small, non-sensitive script and a tiny sample (or synthetic) dataset that reproduces the behavior, plus exact commands and logs.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). In your case (metrics improved but video inference looks worse), please also include your exact inference command/settings and any pre/post-processing details you’re using so we can compare apples-to-apples 🧩.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can try running inference with `end2end=False`

**Comment by Abdullah-Ory** [comment]
> You can try running inference with `end2end=False`

It's worked with me ! thank you, but what does it mean or purpose of it? and does it effect on the inference time ?

**Comment by Y-T-G** [comment]
It makes postprocessing slower

**Comment by Abdullah-Ory** [comment]
> It makes postprocessing slower

Is there a way that i can have the same performance but without setting end2nd = False?

**Comment by Abdullah-Ory** [comment]
> It makes postprocessing slower

Is there a way that i can have the same performance but without setting end2nd = False?

**Comment by Y-T-G** [comment]
No

**Comment by Y-T-G** [comment]
It should only be slightly slower

**Comment by Y-T-G** [comment]
It should only be slightly slower
